### PR TITLE
dav principal search to honour sharing.maxAutocompleteResults setting

### PR DESCRIPTION
**Pull request by blizzz**
This crossed me when investigating a different issue. When using the calendar app to share the calendar, the DAV search is being used. There is not limit, and if I remember correctly the DAV protocol does not foresee to submit a value (correct me if i am wrong). When typing 'pow' in the search field, first I thought I ended up in an infinite loop (for some other changes i did). But in fact, it just searches the LDAP until the very end, resulting in 438 users in the results. 

The limitation speeds up finding users and groups, increasing response time for end users and decreasing load on the servers.

Question: for now I left the default behaviour as unlimited, but would prefer to limit it to 25 as it is done in most other places. Objections? And, are there any fundamental objects to this change?

### CODE REVIEW

File: apps/dav/lib/RootCollection.php [hunk 1]
```diff
@@ -72,7 +72,7 @@ public function __construct() {
 			$proxyMapper,
 			\OC::$server->getConfig()
```
Comment (nickvergessen) [inline apps/dav/lib/RootCollection.php:73]:
```suggestion
			$config
```

File: apps/dav/lib/DAV/GroupPrincipalBackend.php [hunk 2]
```diff
@@ -205,10 +212,11 @@ public function searchPrincipals($prefixPath, array $searchProperties, $test = '
 			$restrictGroups = $this->groupManager->getUserGroupIds($user);
 		}
 
+		$searchLimit = $this->config->getSystemValue('sharing.maxAutocompleteResults', null);
```
Comment (nickvergessen) [inline apps/dav/lib/DAV/GroupPrincipalBackend.php:215]: cast to int|null, just to be sure?

File: apps/dav/lib/RootCollection.php (hunk 1 above)
Comment (blizzz) [reply apps/dav/lib/RootCollection.php:73]: And the userSession further up. And then using `get` over the deprecated getters. And then DI, at least the container. I agree, but that's better kept in a separate PR. I am afraid of an endless rat-tail here.

File: apps/dav/lib/DAV/GroupPrincipalBackend.php (hunk 2 above)
Comment (blizzz) [reply apps/dav/lib/DAV/GroupPrincipalBackend.php]: oh, there's gold:

interface: 

Screenshot_20201211_234750

local:

Screenshot_20201211_235042

I set 25 as default and went with int and a sanitization, put it into a constants, streamlined other differing default values, updated the config.sample.php.

**Review by nickvergessen — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**